entity-service-impl/src/main/java/org/hypertrace/entity/data/service/EntityDataServiceImpl.java

```diff
@@ -694,6 +720,58 @@ private JSONDocument convertEntityRelationshipToDocument(
     }
   }
 
+  private void sentDeleteNotification(String tenantId, java.util.Collection<Entity> entities) {
+    try {
+      entityChangeEventGenerator.sendDeleteNotification(entities);
+    } catch (Exception ex) {
+      LOG.warn(
+          "For tenantId {] unable to send delete notification for entities {}",
+          tenantId,
+          getEntityIds(entities));
+    }
+  }
+
+  private void sendChangeNotification(
+      String tenantId,
+      java.util.Collection<Entity> existingEntities,
+      java.util.Collection<Entity> updatedEntities) {
+    try {
+      entityChangeEventGenerator.sendChangeNotification(existingEntities, updatedEntities);
+    } catch (Exception ex) {
+      LOG.warn(
+          "For tenantId {} unable to send change notification for entities - previous set {} and latest set",
+          tenantId,
+          getEntityIds(existingEntities),
+          getEntityIds(updatedEntities));
+    }
+  }
+
+  private List<String> getEntityIds(java.util.Collection<Entity> entities) {
+    return entities.stream().map(Entity::getEntityId).collect(Collectors.toList());
+  }
+
+  private java.util.Collection<Entity> getExistingEntities(java.util.Collection<Entity> entities) {
```

Comment:
nit: makes no real difference here but... if we have a list, might as well return a list - it can still be used as a collection (accept liberally, return conservatively)

Reply:
done